spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java

```diff
@@ -514,6 +517,20 @@ public void setSize(Integer size) {
 		@NestedConfigurationProperty
 		private final ListenerRetry retry = new ListenerRetry();
 
+		@NestedConfigurationProperty
+		private final SimpleContainer simple = new SimpleContainer();
```

Comment:
OK I get it now. The problem is that there are other nested namespaces that are common. So you have `retry` that is, and `simple` and `direct, at the same level, that are specific. 

I indeed prefer the hierarchical approach but the hierarchy is kind of wrong if you ask me. I have no idea how to fix it though.